docs/src/reference/message-correlation/message-correlation.md

```diff
@@ -0,0 +1,110 @@
+# Message Correlation
+
+Message correlation describes how a message is correlated to a workflow instance. Messages can be correlated to the following elements:
+
+* [Message Start Events](/bpmn-workflows/message-events/message-events.md#message-start-events)
+* [Intermediate Message Catch Events](/bpmn-workflows/message-events/message-events.md#intermediate-message-catch-events)
+* [Message Boundary Events](/bpmn-workflows/message-events/message-events.md#message-boundary-events)
+* [Message Event Subprocesses](/bpmn-workflows/event-subprocesses/event-subprocesses.md)
+* [Receive Tasks](/bpmn-workflows/receive-tasks/receive-tasks.md)
+
+## Message Subscriptions
+
+A message is not sent to a workflow instance directly. Instead, the message correlation is based on subscriptions that contains the **message name** and the **correlation key** (aka correlation value).
+
+![Message Correlation](/reference/message-correlation/message-correlation.png)
+
+A subscription is opened when a workflow instance awaits a message, for example, when entering a message catch event. The message name is defined statically in the workflow (e.g. `Money collected`). The correlation key is defined dynamically as reference of a workflow instance variable (e.g. `orderId`). The value of the variable is used as correlation key of the subscription (e.g. `"order-123"`).
+
+ When a message is published and the message name and the correlation key matches to a subscription then the message is correlated to the corresponding workflow instance. If no proper subscription is opened then the message is discarded.
+
+A subscription is closed when the corresponding element (e.g. the message catch event), or its scope is left. After a subscription is opened, it is not updated, for example, when the referenced workflow instance variable is changed.
+
+<details>
+   <summary>Publish message via zbctl</summary>
+   <p>
+
+   ```
+   zbctl publish message "Money collected" --correlationKey "order-123"
+   ```
+
+   </p>
+ </details>
+
+## Message Cardinality
+
+A message is correlated only **once** to a workflow (based on the BPMN process id), across all versions of this workflow. If multiple subscriptions for the same workflow are opened (by multiple workflow instances or within one instance) then the message is correlated only to one of the subscriptions.
+
+When subscriptions are opened for different workflows then the message is correlated to **all** of the subscriptions.
+
+A message is **not** correlated to a message start event subscription if an instance of the workflow is active and was created by a message with the same correlation key. If the message is buffered then it can be correlated after the active instance is ended. Otherwise, it is discarded.
```

Comment:
To me this is clear, but I already know the feature - I think this might be confusing to some people, maybe some examples would help clarify the concept?

Reply:
Do you mean the part that a message is not correlated to the start event?
This part is also described in the message start event section. I don't have a simple example in mind. 

Do you have an idea for the example?

Reply:
I think the wording is quite clear, but it might not be obvious how it works in practice, and some examples might help people better understand what to expect. I was thinking maybe if we can show them some concrete examples of how this work in practice, it would help - say, given a workflow, and some messages in a particular order, how the execution will look like.

It was just a thought, maybe I'm not giving enough credit to our users and it's not so hard to get your head around the concept :sweat_smile:

Reply:
:sweat_smile: Let's see. If we see users asking for it then I will add an example. Maybe, the upcoming blog post will provide a good example :eyes: